runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp

```diff
@@ -1746,64 +1836,168 @@ static void buildInterfaceCall(TR::CodeGenerator *cg, TR::Node *callNode, TR::Re
+         if ((prevCursor1 != cursor1) || (prevCursor2 != cursor2))
             {
-            debugObj->addInstructionComment(branchToSlot2MissedLabelInstr, "Jumps to slot2MissedLabel");
-            debugObj->addInstructionComment(cursor, "Jumps to slot2DoneLabel");
-            debugObj->addInstructionComment(slot2MissedLabelInstr, "slot2MissedLabel");
-            debugObj->addInstructionComment(slot2DoneLabelInstr, "slot2DoneLabel");
+            TR::LabelSymbol *lastITableMissedLabel = generateLabelSymbol(cg);
+            TR::LabelSymbol *lastITableDoneLabel = generateLabelSymbol(cg);
+            /* Debug counter was generated. Generating instructions before debug counter instructions recording hit for lastITable cache. */
+            TR::Instruction *branchToLastITableMissedLabelInstr = generateConditionalBranchInstruction(cg, TR::InstOpCode::b_cond, callNode, lastITableMissedLabel, TR::CC_NE, prevCursor1);
+
+            /* Generating instructions before debug counter instructions recording cache miss. */
+            TR::Instruction *cmpInstr1 = generateCompareInstruction(cg, callNode, x9Reg, x9Reg, true, prevCursor2); /* to set Z flag */
+            TR::Instruction *cursor = generateLabelInstruction(cg, TR::InstOpCode::b, callNode, lastITableDoneLabel, cmpInstr1);
+            TR::Instruction *lastITableMissedLabelInstr = generateLabelInstruction(cg, TR::InstOpCode::label, callNode, lastITableMissedLabel, cursor);
+
+            /* Generating instructions after debug counter instructions. */
+            generateCompareImmInstruction(cg, callNode, x10Reg, 0, true); /* to unset Z flag */
+            TR::Instruction *lastITableDoneLabelInstr = generateLabelInstruction(cg, TR::InstOpCode::label, callNode, lastITableDoneLabel);
+            if (debugObj)
+               {
+               debugObj->addInstructionComment(branchToLastITableMissedLabelInstr, "Jumps to lastITableMissedLabel");
+               debugObj->addInstructionComment(cursor, "Jumps to lastITableDoneLabel");
+               debugObj->addInstructionComment(lastITableMissedLabelInstr, "lastITable2MissedLabel");
+               debugObj->addInstructionComment(lastITableDoneLabelInstr, "lastITableDoneLabel");
+               }
             }
          }
-      srm->stopUsingRegisters();
+
+      /* This conditional branch instruction with "always" condition code will be patched to b.ne instruction after second cache slot is filled. */
+      gcPoint = generateConditionalBranchInstruction(cg, TR::InstOpCode::b_cond, callNode, ifcSnippetLabel, TR::CC_AL);
+      loadConstant32(cg, callNode, fej9->getITableEntryJitVTableOffset(), x9Reg);
+      generateTrg1MemInstruction(cg, TR::InstOpCode::ldrimmx, callNode, x11Reg, TR::MemoryReference::createWithDisplacement(cg, x10Reg, fej9->convertITableIndexToOffset(itableIndex)));
+      generateTrg1Src2Instruction(cg, TR::InstOpCode::subx, callNode, x9Reg, x9Reg, x11Reg);
```

Comment:
Please add a comment here: `// PicBuilder.spp checks this instruction.  It needs to be 'sub x9, x9, x11'`.

Reply:
Added a comment.